### ISSUE TITLE
excluding entity warnings from logger when source is homeassistant.helpers.entity

### ISSUE THREAD

**Issue by bartgrefte** [home-assistant/home-assistant.github.io#20566]
### Feedback

HASS has been giving lot's of warnings about various devices/entities, primarily because of connection issues due to those devices being in sleep mode or turned off. I managed to get rid of most warnings using the info on https://www.home-assistant.io/integrations/logger/ , unfortunately there's info missing on that page. 

One of the warnings does not have the device's entity as source, instead it's this:

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:661
First occurred: 2 december 2021 01:06:19 (216 occurrences)
Last logged: 06:24:24
Update of media_player.enigma2_media_player is taking over 10 seconds

Please update https://www.home-assistant.io/integrations/logger/ with info to exclude warnings from entities (in this case media_player.enigma2_media_player) with helpers.entity as source. This because adding media_player.enigma2_media_player like below does not prevent the warning mentioned above:

```
logger:
  default: warning
  logs:
    openwebif.api: fatal
    media_player.enigma2_media_player: fatal
    homeassistant.components.media_player: fatal
    homeassistant.components.brother: fatal
    pychromecast.socket_client: fatal
```

I also have a feature request with regarding to logger. Please add the option to exclude certain log entries, like connection related warnings, based on time. To stay with the same example (media_player.enigma2_media_player) this device has been configured to switch itself completely off at 1 AM and switch itself back on at 6.30AM.

During this time period the log is flooded with connection warnings and lack of update warnings (like above), it would help to have an option that can prevent flooding of specific log entries from specific entities during a preset time period.

### URL

https://www.home-assistant.io/integrations/logger/

### Version

2021.11.5

**Comment by balloob** [comment]
The solution here is not to make it easier for people to hide the warnings, but to fix the warnings in the first place. If an update takes over 10 seconds, it's probably unreachable and we should raise an error or the polling inerval needs to be extended.

**Comment by bartgrefte** [comment]
@balloob Is extending the polling interval something that can be done manually or is there no support at all for something like that? In my case adding the option for a specific time interval to not log certain warnings would help, at least with one device (enigma2).

"should raise an error"
A connection error when the device is in sleep/shutdown? That's the main reason I'm getting connection warnings, what would be the point in raising an error instead of a warning, wouldn't that still flood the log?

While waiting for a solution, I have added helpers/entity.py:661 to the logger config, hoping that will finally get rid of the warning I listed in the first post, hopefully that only affects that ( Update of media_player.enigma2_media_player is taking over 10 seconds) warning and no others, no idea if helpers/entity.py:661 is used for warnings of other devices.

The same goes for config_entries.py:335, which gives connection warnings about the Brother printer even though I have added "homeassistant.components.brother: fatal" to the config..

**edit:** config_entries.py:335: fatal has no effect, still getting warnings from the Brother printer.
